lib/internal/modules/esm/resolve.js

```diff
@@ -1141,7 +1139,13 @@ async function defaultResolve(specifier, context = {}, defaultResolveUnused) {
   if (maybeReturn) return maybeReturn;
 
   // This must come after checkIfDisallowedImport
-  if (parsed && parsed.protocol === 'node:') return { url: specifier };
+  if (parsed) {
+    if (NativeModule.canBeRequiredByUsers(parsed.pathname)) { // maybe redundant
```

Comment:
this should only be done for `node:` right now `foo:fs` triggers this.

Reply:
that sounds like these are redundant then (or should be consolidated), no?